test/integration/targets/vmware_guest_sendkey/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+# Test code for the vmware_guest_sendkey module
+# Copyright: (c) 2017, Diane Wang (Tomorrow9) <dianew@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- when: vcsim is not defined
+  block:
+  - import_role:
+      name: prepare_vmware_tests
+    vars:
+      setup_attach_host: true
+      setup_datastore: true
+      setup_virtualmachines: true
+
+  - name: create a new virtual machine and power on
+    vmware_guest:
+      validate_certs: False
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      folder: "vm"
+      name: Test-VM
+      datacenter: "{{ dc1 }}"
+      cluster: "{{ ccr1 }}"
+      resource_pool: Resources
+      guest_id: centos64Guest
+      state: poweredon
+      hardware:
+        memory_mb: 512
+        num_cpus: 1
+        scsi: paravirtual
+      disk:
+      - size_mb: 128
+        type: thin
+        datastore: '{{ ds2 }}'
+      cdrom:
+        type: client
+    register: test_vm
+
+  - debug: var=test_vm
+
+  - name: assert the VM was created
+    assert:
+      that:
+        - "test_vm.failed == false"
+        - "test_vm.changed == true"
+
+  - name: send keys to created virtual machine
+    vmware_guest_sendkey:
+      validate_certs: False
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      folder: "vm"
+      name: Test-VM
+      datacenter: "{{ dc1 }}"
+      cluster: "{{ ccr1 }}"
+      keys_send:
+        - DOWNARROW
+        - DOWNARROW
+        - ENTER
+      string_send: "test-user"
+    register: send_key
+
+  - debug: var=send_key
+
+  - name: assert the keys were sent to VM
+    assert:
+      that:
+        - "send_key.failed == false"
+        - "send_key.changed == true"
```

Comment:
You don't need to check the `failed` here since the playbook will stop before in this case anyway.
Do 'send_key is changed' instead

Reply:
Modified. Thanks.